utils/build_swift/build_swift/driver_arguments.py

```diff
@@ -246,7 +246,11 @@ def _apply_default_arguments(args):
         args.test_watchos_simulator = False
 
     if not args.build_android:
-        args.test_android = False
+        # If building natively on an Android host, allow running the test suite
+        # without the NDK config.
+        if not StdlibDeploymentTarget.Android.contains( \
+                   StdlibDeploymentTarget.host_target().name):
+            args.test_android = False
```

Comment:
>a lot of logic is tied to args.android, which at this point means "cross compile Android with the NDK", but having this args.build_android being false for native Android builds is just plain confusing.

I found it confusing that there's a pair of seemingly redundant booleans to enable building for each platform, ie `android`/`build_android` or `ios`/`build_ios`, but I'm guessing it has to do with options parsing so that you can pass both `--android` and `--skip-build-android` and not have weird races or ordering issues.

You are right that it's confusing that these variables really mean the Android NDK config, but I tried to clear that up with the comments and I doubt anyone will touch the native host config after this, so probably fine.

If you have some suggestion to clarify it by internal variable renaming or something, e.g. `android` becomes `android_ndk`, just let me know.

>otherwise the test_android_host = False still happens in native compilation, and that means that the executable tests will not happen by default.

No, that was intentional, I don't want the host tests with adb to run. if `android_host_test` were set to true, _that's_ when the executable tests are disabled, not the other way around. Maybe you're thinking of some older config in `build-script-impl`, but that's gone now.